lib/internal/streams/operators.js

```diff
@@ -108,27 +131,28 @@ function map(fn, options) {
             throw new AbortError();
           }
 
+          if (val === kEmpty) {
```

Comment:
This might be in the wrong place. I think it should be after call to fn

Reply:
I'm surprised no tests caught

Reply:
no test caught it as it's an optimization rather than a guard, the real guard is here:
https://github.com/nodejs/node/blob/c0951c7787cc9d5a3cce877151099ab2d3f009e3/lib/internal/streams/operators.js#L191-L193

Reply:
moved back anyway